File: templates/nodejs-service/service/context/context.js
const { Config } = require("../config")

class Context {

    constructor() {
        if (!("status" in Context)) {
            Context.status = {
                isAlive: true,
                isReady: true
            };
        }
        if (!("localCache" in Context)) {
            Context.localCache = {};
        }
        if (!("config" in Context)) {
            Context.config = new Config();
        }
    }

    initLocalCache(key, val) {
        if (!(key in Context.localCache)) {
            this.setLocalCache(key, val);
        }
        return this;
    }

    setLocalCache(key, val) {
        Context.localCache[key] = val;
        return this;
    }

    getLocalCache(key) {
        return Context.localCache[key];
    }

    getConfig() {
        return Context.config;
    }

    getLiveness() {
        return Context.status.isAlive;
    }

    setLiveness(liveness) {
        Context.status.isAlive = liveness;
    }

    getReadiness() {
        return Context.status.isReady;
    }

    setReadiness(readiness) {
        Context.status.isReady = readiness;
    }

}

module.exports = {
    Context
};

File: templates/go-service/service/context/context.go
package context

import (
	"app/config"
	"app/transport"
)

// NewContext initiate new Context
func NewContext() (context *Context) {
	return &Context{
		Config: config.NewConfig(),
		Status: &Status{
			IsAlive: true,
			IsReady: true,
		},
		LocalCache: transport.Message{},
	}
}

// Context is a general service context
type Context struct {
	Config     *config.Config
	Status     *Status
	LocalCache transport.Message
}

// InitLocalCache set LocalCache
func (context *Context) InitLocalCache(key string, val interface{}) {
	if _, exists := context.LocalCache[key]; !exists {
		context.LocalCache[key] = val
	}
}

// Status represent liveness and readiness of service
type Status struct {
	IsAlive bool
	IsReady bool
}
